lib/SILOptimizer/Analysis/ClassHierarchyAnalysis.cpp

```diff
@@ -32,7 +32,7 @@ class NominalTypeWalker: public ASTWalker {
 
   bool walkToDeclPre(Decl *D) override {
     auto *NTD = dyn_cast<NominalTypeDecl>(D);
-    if (!NTD)
+    if (!NTD || !NTD->hasInterfaceType())
```

Comment:
@slavapestov If `!NTD->hasInterfaceType()` is true, does it stay like this for the whole duration of compilation or may it become false at some point? I.e. may this NTD become a class that we should have taken into account?